plugins/sparc_engage_warn.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# Shows a warning message before engage the SPARC mirror
+'''
+Created on 3 Feb 2021
+
+@author: Éric Piel
+Copyright © 2021 Éric Piel, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms of the GNU
+General Public License version 2 as published by the Free Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis. If not,
+see http://www.gnu.org/licenses/.
+'''
+
+from __future__ import division
+
+import logging
+from odemis.gui.cont.tabs import MIRROR_PARKED
+from odemis.gui.plugin import Plugin
+import wx
+
+ENGAGE_MSG = """Are you sure you want to engage the mirror?
+* Ensure stage is at a working distance of at least 18 mm.
```

Comment:
Is there now way of checking what the current working distance is and including that in the message?

Reply:
Nope. We often don't have access to the stage, and even more rarely have the information on the working distance.